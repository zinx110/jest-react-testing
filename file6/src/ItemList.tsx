type itemType = {
    name: string;
    id: string;
};
type itemsType = Array<itemType>;

interface ItemListProps {
    items: itemsType;
}

const ItemList = ({ items }: ItemListProps) => {
    return (
        <ul>
            {items.map((item: itemType) => (
                <li key={item.id}>{item.name}</li>
            ))}
        </ul>
    );
};

export default ItemList;
