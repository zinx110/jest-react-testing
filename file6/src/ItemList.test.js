import React from "react";
import TestRenderer from "react-test-renderer";
import ItemList from "./ItemList";

describe("ItemList", () => {
    const items = [
        { id: 1, name: "Item 1" },
        { id: 2, name: "Item 2" },
        { id: 3, name: "Item 3" },
    ];
    it("renders correctly", () => {
        const itemListComp = TestRenderer.create(<ItemList items={items} />);

        expect(itemListComp.toJSON()).toMatchSnapshot();
    });
    it("renders the correct number of items", () => {
        const itemListComp = TestRenderer.create(<ItemList items={items} />);
        const root = itemListComp.root;
        expect(root.findAllByType("li").length).toEqual(items.length);
    });
    xit("Test skip", () => {});
    it.todo("Test todo");

    it("Renders the correct item names", () => {
        const itemListComp = TestRenderer.create(<ItemList items={items} />);
        const root = itemListComp.root;
        const liNodes = root.findAllByType("li");
        liNodes.forEach((node, index) => {
            expect(node.children).toEqual([items[index].name]);
        });
    });
});
