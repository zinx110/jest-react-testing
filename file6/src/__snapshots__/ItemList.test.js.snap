// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList renders correctly 1`] = `
<ul>
  <li>
    Item 1
  </li>
  <li>
    Item 2
  </li>
  <li>
    Item 3
  </li>
</ul>
`;
